components/admin_console/users_and_teams_settings/users_and_teams_settings.jsx

```diff
@@ -2,47 +2,107 @@
 // See License.txt for license information.
 
 import React from 'react';
+import PropTypes from 'prop-types';
 import {FormattedHTMLMessage, FormattedMessage} from 'react-intl';
 
+import {RequestStatus} from 'mattermost-redux/constants';
+
 import Constants from 'utils/constants.jsx';
 import * as Utils from 'utils/utils.jsx';
+import {rolesFromMapping, mappingValueFromRoles} from 'utils/policy_roles_adapter';
+
+import LoadingScreen from 'components/loading_screen.jsx';
 
-import AdminSettings from './admin_settings.jsx';
-import BooleanSetting from './boolean_setting.jsx';
-import DropdownSetting from './dropdown_setting.jsx';
-import SettingsGroup from './settings_group.jsx';
-import TextSetting from './text_setting.jsx';
+import AdminSettings from '../admin_settings.jsx';
+import BooleanSetting from '../boolean_setting.jsx';
+import DropdownSetting from '../dropdown_setting.jsx';
+import SettingsGroup from '../settings_group.jsx';
+import TextSetting from '../text_setting.jsx';
 
 const RESTRICT_DIRECT_MESSAGE_ANY = 'any';
 const RESTRICT_DIRECT_MESSAGE_TEAM = 'team';
 
 export default class UsersAndTeamsSettings extends AdminSettings {
+    static propTypes = {
+        actions: PropTypes.shape({
+            loadRolesIfNeeded: PropTypes.func.isRequired,
+            editRole: PropTypes.func.isRequired
+        }).isRequired
+    };
+
     constructor(props) {
         super(props);
 
         this.getConfigFromState = this.getConfigFromState.bind(this);
```

Comment:
We can convert it to an arrow function and remove this line. (Not directly related to the PR)